cluster-operator/src/test/java/io/strimzi/operator/cluster/model/KafkaClusterTest.java

```diff
@@ -616,6 +619,64 @@ public void testExternalNodePorts() {
         }
     }
 
+    @Test
+    public void testExternalNodePortOverrides() {
+        Kafka kafkaAssembly = new KafkaBuilder(ResourceUtils.createKafkaCluster(namespace, cluster, replicas,
+            image, healthDelay, healthTimeout, metricsCm, configuration, emptyMap()))
+            .editSpec()
+                .editKafka()
+                    .withNewListeners()
+                        .withNewKafkaListenerExternalNodePort()
+                            .withTls(false)
+                            .withOverrides(new KafkaExternalServiceOverrides(new KafkaExternalBootstrapService(10),
+                                Collections.singletonList(new KafkaExternalBrokerService(0, "test", 0, 100))))
+                        .endKafkaListenerExternalNodePort()
+                    .endListeners()
+                .endKafka()
+            .endSpec()
+            .build();
+        KafkaCluster kc = KafkaCluster.fromCrd(kafkaAssembly, VERSIONS);
+
+        SortedMap<Integer, String> addresses = new TreeMap<>();
+        addresses.put(0, "32123");
```

Comment:
We are testing the nodeport override for pod 0. We want to see that pod 0 will have nodeport 100 instead of the one set here.
I will change the port to more realistic number (let's say 32101), same for advertisedPort.

Reply:
Right. But if the override specifies that the nodeport number should be `32101`, then it will be `32101`. Instead in this test case, you seem to request the node port to be `32101` but based on what you pass back in this Map - which contains just the list of used node ports -  it seems to be `32123`. So that doesn't really match with what you expect to happen when you request nodeport `32101`.